Source/Orts.Simulation/Simulation/Signalling/Signals.cs

```diff
@@ -12130,7 +12130,9 @@ public bool TrainHasCallOn(bool allowOnNonePlatform, string dumpfile)
             }
 
             // signal is not first signal for train
-            if (enabledTrain.Train.NextSignalObject[enabledTrain.TrainRouteDirectionIndex] != null &&
+
+            if (!allowAdvancedSignal &&
+                enabledTrain.Train.NextSignalObject[enabledTrain.TrainRouteDirectionIndex] != null &&
                 enabledTrain.Train.NextSignalObject[enabledTrain.TrainRouteDirectionIndex].thisRef != thisRef)
```

Comment:
```suggestion
                enabledTrain.Train.NextSignalObject[enabledTrain.TrainRouteDirectionIndex]?.thisRef != thisRef)
```
Simplify null checks with an optional.